napalm/nxos/nxos.py

```diff
@@ -779,10 +779,20 @@ def get_facts(self):
         facts = {}
         facts["vendor"] = "Cisco"
 
+        show_inventory_table = self._get_command_table(
+            "show inventory", "TABLE_inv", "ROW_inv"
+        )
+        if isinstance(show_inventory_table, dict):
+            show_inventory_table = [show_inventory_table]
+
+        for row in show_inventory_table:
+            if row["name"] == "Chassis":
+                facts["serial_number"] = row.get("serialnum", "")
+                break
+
         show_version = self._send_command("show version")
         facts["model"] = show_version.get("chassis_id", "")
         facts["hostname"] = show_version.get("host_name", "")
-        facts["serial_number"] = show_version.get("proc_board_id", "")
```

Comment:
I'd say that `serial_number` should always be there, while the loop above presents the risk of returning an output that doesn't contain it. Might want to add a line `facts['serial_number'] = None` just above maybe?

Reply:
Added to code